### DISCUSSION TITLE
How Can I Visualize Multiple SWATHobj Profiles on a Single Plot in TopoToolbox?

### DISCUSSION THREAD

**Discussion by taylansancar** [wschwanghart/topotoolbox#50] [Questions]
![image](https://github.com/user-attachments/assets/5b314a69-d5d9-419f-9cb8-75fc7e0e84ae)

<img width="468" alt="image" src="https://github.com/user-attachments/assets/259ffdc1-590a-4db5-81fa-6705e1b1d260">

**Comment by wschwanghart** [comment]
Hi, 
that's weird, I can run `SWATHobj/plotdz` and plot various swath profiles into one axes. Which version are you using?
Best, Wolfgang

**Reply by taylansancar** [reply]
Hi,
Version:TopoToolbox 2.4

**Reply by taylansancar** [reply]
Hi Wolfgang,

I wanted to apologize for the confusion caused by my question yesterday about the TopoToolbox code. I later realized that the issue was due to a compatibility problem with MATLAB on my MacBook, and the code (below) actually worked perfectly on Windows. I’m sorry for taking up your time with this, and I appreciate your interest.

Thank you for your help and understanding!


clear
clc
close all
addpath(genpath('E:\topotoolbox'));

DEM = GRIDobj('sasSwath.tif'); 

SW1 = SWATHobj(DEM, 'width', 500);
SW2 = SWATHobj(DEM, 'width', 500);

figure()
hold on

plotdz(SW1, 'boundedline', true, 'facecolor', [0.3 0.5 1], 'minmaxfacecolor', [0.7 0.8 1], ...
    'facealpha', 0.8, 'minmaxfacealpha', 0.4);

plotdz(SW2, 'boundedline', true, 'facecolor', [1 0.5 0.5], 'minmaxfacecolor', [1 0.8 0.8], ...
    'facealpha', 0.4, 'minmaxfacealpha', 0.2);

hold off
title('Swath Profiles (')
xlabel('Distance (m)')
ylabel('Elevation (m)')
legend('S Profile 1', 'S Profile 2')

**Reply by wschwanghart** [reply]
That's totally fine. Always glad to hear that a problem is solved ;-)
